File: test/python/test_header.py
from rixmsg.message import Message
from rixmsg.standard.Header import Header

def print_header(header: Header):
    print(f"seq: {header.seq}")
    print(f"stamp.sec: {header.stamp.sec}")
    print(f"stamp.nsec: {header.stamp.nsec}")
    print(f"frame_id: {header.frame_id}")

header = Header()
header.seq = 123
header.stamp.sec = 456
header.stamp.nsec = 789
header.frame_id = 'Hello, world!'

print("Original:")
print_header(header)
print(f"Original size: {header.size()}")

header_serialized = bytearray()
header.serialize(header_serialized)

print("Serialized:")
print(header_serialized)
print(f"Serialized size: {len(header_serialized)}")

header_deserialized = Header()
header_deserialized.deserialize(header_serialized, Message.Offset())

print("Deserialized:")
print_header(header_deserialized)

File: test/js/test_header.js
import { Header } from 'rixmsg/standard/Header.js';

function print_header(msg) {
    console.log('seq:', msg.seq);
    console.log('stamp.sec:', msg.stamp.sec);
    console.log('stamp.nsec:', msg.stamp.nsec);
    console.log('frame_id:', msg.frame_id);
}

var header = new Header();
header.seq = 0;
header.stamp.sec = 1234;
header.stamp.nsec = 5678;
header.frame_id = 'Hello, world!';

console.log("Original: ");
print_header(header);

let encoded = new ArrayBuffer();
encoded = header.serialize(encoded);

console.log("\nEncoded: ");
console.log(encoded);

let context = { offset: 0 };
let decoded = new Header();
let view = new DataView(encoded);
decoded.deserialize(view, context);

console.log("\nDecoded: ");
print_header(decoded);